### PR TITLE
refactor: [M3-6263] - MUI v5 Migration - SRC > Features > EntityTransfers

### PR DESCRIPTION
**Pull request by tyler-akamai**
## Description 📝
MUI v5 CSS migration for SRC > Features > EntityTransfers

## Major Changes 🔄
- Removed `React.FC` or `
- Renamed interface `Props` to `{ComponentName}Props`.
- Removed `CombinedProps` if it's directly referencing `Props` with nothing additional
- Removed `makeStyles` for new `styled API`.
- `export` component inline and removed `export default {ComponentName}` at bottom of file. 

## Preview 📷
| Before  | After   |
| ------- | ------- |
| Screenshot 2023-08-23 at 4 46 02 PM | Screenshot 2023-08-23 at 4 46 10 PM |
| Screenshot 2023-08-23 at 4 47 10 PM | Screenshot 2023-08-23 at 4 47 22 PM |
| Screenshot 2023-08-23 at 4 47 42 PM | Screenshot 2023-08-23 at 4 48 07 PM |

## How to test 🧪
1. Navigate to http://localhost:3000/account/service-transfers
2. Ensure the functionality and styling is consistent

### CODE REVIEW

File: packages/manager/src/features/EntityTransfers/EntityTransfersCreate/TransferHeader.tsx [hunk 1]
```diff
@@ -2,13 +2,11 @@ import * as React from 'react';
 
 import { Typography } from 'src/components/Typography';
 
-export const TransferHeader: React.FC<{}> = (_) => {
+export const TransferHeader = React.memo(() => {
```
Comment (abailly-akamai) [inline packages/manager/src/features/EntityTransfers/EntityTransfersCreate/TransferHeader.tsx:5]: nitpick - there's nothing to memoize here

File: packages/manager/src/features/EntityTransfers/EntityTransfersCreate/TransferTable.styles.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  '& svg': {
+    height: 20,
+    width: 20,
+  },
+});
+
+export const StyledEmptyCheckbox = styled(Checkbox, {
+  label: 'StyledEmptyCheckbox',
+})({
+  '& svg': { color: '#cccccc', height: 20, width: 20 },
```
Comment (abailly-akamai) [inline packages/manager/src/features/EntityTransfers/EntityTransfersCreate/TransferTable.styles.ts:21]: I'd rather not see hardcoded colors if you can find a suitable variable - usually a red flag for light/dark theme handling

File: packages/manager/src/features/EntityTransfers/EntityTransfersCreate/TransferTable.styles.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      backgroundColor: theme.bg.tableHeader,
+      borderBottom: `2px solid ${theme.borderColors.borderTable}`,
+      borderLeft: `1px solid ${theme.borderColors.borderTable}`,
+      borderRight: `1px solid ${theme.borderColors.borderTable}`,
+      borderTop: `2px solid ${theme.borderColors.borderTable}`,
+      color: theme.textColors.tableHeader,
+      fontFamily: theme.font.bold,
+      fontSize: '0.875em !important',
+      padding: '0px 15px',
+    },
```
Comment (abailly-akamai) [inline packages/manager/src/features/EntityTransfers/EntityTransfersCreate/TransferTable.styles.ts:67]: When possible, try to not to nest (especially 3 levels) and rather make a styled component for the `th` itself

File: packages/manager/src/features/EntityTransfers/EntityTransfersLanding/ConfirmTransferDialog.test.tsx [hunk 4]
```diff
@@ -6,7 +6,8 @@ import { entityTransferFactory } from 'src/factories/entityTransfers';
 import { rest, server } from 'src/mocks/testServer';
 import { renderWithTheme } from 'src/utilities/testHelpers';
 
-import ConfirmTransferDialog, {
+import {
+  ConfirmTransferDialog,
   Props,
```
Comment (abailly-akamai) [inline packages/manager/src/features/EntityTransfers/EntityTransfersLanding/ConfirmTransferDialog.test.tsx:11]: nitpick - when we think about it, let's import types separately with `import type { ... }`

File: packages/manager/src/features/EntityTransfers/EntityTransfersLanding/TransferControls.styles.ts [hunk 5]
```diff
@@ -0,0 +1,133 @@
+import Grid from '@mui/material/Unstable_Grid2';
+import { styled } from '@mui/material/styles';
+
+import { Button } from 'src/components/Button/Button';
+import { TextField } from 'src/components/TextField';
+import { Typography } from 'src/components/Typography';
+
+// sm = 600, md = 960, lg = 1280
+
+const xs_sm = 450;
```
Comment (abailly-akamai) [inline packages/manager/src/features/EntityTransfers/EntityTransfersLanding/TransferControls.styles.ts:10]: nipick: name your constant better (human readable) and uppercase it

File: packages/manager/src/features/EntityTransfers/RenderTransferRow.styles.ts [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+export const StyledTableCell = styled(TableCell, {
+  label: 'StyledTableCell',
+})({
+  paddingLeft: '1rem',
+});
+
+export const StyledTokenTableCell = styled(TableCell, {
+  label: 'StyledTokenTableCell',
+})(({ theme }) => ({
+  paddingLeft: '1rem',
```
Comment (abailly-akamai) [inline packages/manager/src/features/EntityTransfers/RenderTransferRow.styles.ts:16]: try to use theme variables for spacing, esp for round values - true for this file and the rest of the PR

File: packages/manager/src/features/EntityTransfers/RenderTransferRow.styles.ts [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    width: 12,
+  },
+  marginLeft: theme.spacing(1),
+  marginTop: 2,
+}));
+
+export const StyledButton = styled('button', {
+  label: 'StyledButton',
+})(({ theme }) => ({
+  ...theme.applyLinkStyles,
```
Comment (abailly-akamai) [inline packages/manager/src/features/EntityTransfers/RenderTransferRow.styles.ts:60]: use `StyledLinkButton.ts`

File: packages/manager/src/features/Kubernetes/KubernetesClusterDetail/NodePoolsDisplay/NodePoolsDisplay.tsx [hunk 8]
```diff
@@ -87,7 +87,7 @@ export const NodePoolsDisplay = (props: Props) => {
 
   const [isAutoscaleDialogOpen, setIsAutoscaleDialogOpen] = useState(false);
 
-  const [numPoolsToDisplay, setNumPoolsToDisplay] = React.useState(25);
+  const [numPoolsToDisplay, setNumPoolsToDisplay] = React.useState(5);
```
Comment (abailly-akamai) [inline packages/manager/src/features/Kubernetes/KubernetesClusterDetail/NodePoolsDisplay/NodePoolsDisplay.tsx:90]: Thanks for catching this - I think that was a testing change i had left behind 👍

**Review by abailly-akamai — APPROVED** [review]
Looks good! Approving pending some cleanup

**Review by abailly-akamai — COMMENTED** [review]
oops i missed this was a draft - comments still apply tho :)

File: packages/manager/src/features/Kubernetes/KubernetesClusterDetail/NodePoolsDisplay/NodePoolsDisplay.tsx (hunk 8 above)
Comment (tyler-akamai) [reply packages/manager/src/features/Kubernetes/KubernetesClusterDetail/NodePoolsDisplay/NodePoolsDisplay.tsx:90]: oops I didnt mean to add this to the PR, I was testing your scrollable pr in the wrong branch. Do you want me to keep these changes?

File: packages/manager/src/features/EntityTransfers/EntityTransfersLanding/ConfirmTransferDialog.test.tsx [hunk 9]
```diff
@@ -6,11 +6,13 @@ import { entityTransferFactory } from 'src/factories/entityTransfers';
...
 import { renderWithTheme } from 'src/utilities/testHelpers';
 
-import ConfirmTransferDialog, {
-  Props,
+import {
+  ConfirmTransferDialog,
   getTimeRemaining,
 } from './ConfirmTransferDialog';
 
+import type { Props } from './ConfirmTransferDialog';
```
Comment (coliu-akamai) [inline packages/manager/src/features/EntityTransfers/EntityTransfersLanding/ConfirmTransferDialog.test.tsx:14]: nitpick / feel free to ignore -- if we're importing/exporting Props, might be worth making the naming more specific to ConfirmTransferDialogProps or something like that (although this is just the test file for the component, so not a big deal)

File: packages/manager/src/features/EntityTransfers/EntityTransfersCreate/TransferTable.styles.ts [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+})(({ theme }) => ({
+  marginBottom: theme.spacing(),
+  marginTop: theme.spacing(2),
+}));
+
+export const StyledDebouncedSearchTextField = styled(DebouncedSearchTextField, {
+  label: 'StyledDebouncedSearchTextField',
+})(({ theme }) => ({
+  marginBottom: theme.spacing(0.5),
+  maxWidth: 556,
```
Comment (coliu-akamai) [inline packages/manager/src/features/EntityTransfers/EntityTransfersCreate/TransferTable.styles.ts:43]: 😆 I wonder how they got this specific of a number

File: packages/manager/src/features/EntityTransfers/EntityTransfersLanding/TransferControls.styles.ts [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+    flexWrap: 'wrap',
+  },
+  [theme.breakpoints.down('md')]: {
+    '& > div': {
+      margin: 0,
+      padding: 0,
+    },
+    marginBottom: theme.spacing(1),
+    padding: 0,
+    width: 'calc(600px - 32px)',
```
Comment (coliu-akamai) [inline packages/manager/src/features/EntityTransfers/EntityTransfersLanding/TransferControls.styles.ts:40]: nitpick: could we just simplify this to be `width: '568px`'?

File: packages/manager/src/features/EntityTransfers/EntityTransfersLanding/TransferControls.styles.ts [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+  },
+}));
+
+export const StyledTransferGrid = styled(Grid, {
+  label: 'StyledTransferGrid',
+})(({ theme }) => ({
+  paddingLeft: 0,
+  paddingRight: 0,
+  [theme.breakpoints.down('md')]: {
+    width: 'calc(600px - 32px)',
```
Comment (coliu-akamai) [inline packages/manager/src/features/EntityTransfers/EntityTransfersLanding/TransferControls.styles.ts:58]: same here

File: packages/manager/src/features/EntityTransfers/EntityTransfersLanding/TransferControls.styles.ts [hunk 13]
```diff
@@ -0,0 +1,134 @@
...
+
+export const StyledTransferWrapperGrid = styled(Grid, {
+  label: 'StyledTransferWrapperGrid',
+})(({ theme }) => ({
+  [theme.breakpoints.down('md')]: {
+    '& > div': {
+      flexGrow: 1,
+    },
+    flexBasis: '100%',
+    width: 'calc(600px-32px)',
```
Comment (coliu-akamai) [inline packages/manager/src/features/EntityTransfers/EntityTransfersLanding/TransferControls.styles.ts:129]: and here as well

**Review by coliu-akamai — COMMENTED** [review]
When there are no linodes in the /account/service-transfers/create page, the table column widths are different on your branch vs prod:

branch:
image

prod:
image

Also, it's not really easy to screenshot so I've only included an image of the area, not a comparison, but the Service Transfer Summary area on the same page may be slightly shifted? Will say though that sometimes I've come across spacing issues like this that just... disappeared the next day (???) so it may not be anything, but worth double checking: 
image

File: packages/manager/src/features/EntityTransfers/EntityTransfersCreate/TransferCheckoutBar.styles.ts [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+
+import { Button } from 'src/components/Button/Button';
+import { Typography } from 'src/components/Typography';
+
+export const StyledButton = styled(Button, {
+  label: 'StyledButton',
+})(({ theme }) => ({
+  backgroundColor: 'inherit',
+  border: 'none',
+  minWidth: 0,
```
[the remark this answers is not in the document]
Comment (tyler-akamai) [reply packages/manager/src/features/EntityTransfers/EntityTransfersCreate/TransferCheckoutBar.styles.ts:12]: correct, what matters is the Styled Closed button

File: packages/manager/src/features/Kubernetes/KubernetesClusterDetail/NodePoolsDisplay/NodePoolsDisplay.tsx (hunk 8 above)
Comment (abailly-akamai) [reply packages/manager/src/features/Kubernetes/KubernetesClusterDetail/NodePoolsDisplay/NodePoolsDisplay.tsx:90]: yes please!

**Review by coliu-akamai — COMMENTED** [review]
Still getting a slight difference in width if there are no linodes for the /service-transfers/create page on this branch vs prod, seems to be the checkbox column. Other than that everything else looked good!

image

prod
image

**Comment by tyler-akamai** [comment]
> Still getting a slight difference in width if there are no linodes for the /service-transfers/create page on this branch vs prod, seems to be the checkbox column. Other than that everything else looked good!
> 
> image
> 
> prod image

Yeah its a little wider without data, but as soon as data is added, it expands to the width I currently have. Its a minor difference but I wouldn't consider it a regression.

**Review by coliu-akamai — APPROVED** [review]
Ok thanks for the clarification, approving in that case 🎉

File: packages/manager/src/features/EntityTransfers/EntityTransfersLanding/ConfirmTransferDialog.tsx [hunk 15]
```diff
@@ -30,144 +26,127 @@ import { formatDate } from 'src/utilities/formatDate';
...
+          queryClient.invalidateQueries({
+            predicate: (query) =>
+              query.queryKey[0] === queryKey &&
+              query.queryKey[2] === TRANSFER_FILTERS.received,
+          });
+          onClose();
+          setSubmitting(false);
+          enqueueSnackbar('Transfer accepted successfully.', {
+            variant: 'success',
+          });
```
Comment (cpathipa) [inline packages/manager/src/features/EntityTransfers/EntityTransfersLanding/ConfirmTransferDialog.tsx:105]: Nice to have: we could simplify this with async/await's

**Review by cpathipa — COMMENTED** [review]
Looks like search functionality in Make a Service Transfer page is not working as I expected.

| Local | Production |
|-----------------|-----------------|
| image | image |

**Comment by coliu-akamai** [comment]
@cpathipa if you're using the MSW, I'm not too sure if the search bar filters the MSW linodes: I'm getting the same behavior on the develop branch, where I can't search/filter MSW linodes. However, I tried turning off the MSW and creating some linodes, and the searchbar seemed to work fine for the actual linodes -- could you confirm this behavior on your end?

**Pull request merged**